### PR TITLE
virtual attribute processing added to LDAPc

### PR DESCRIPTION
**Pull request by mvocu**
- added beans to describe virtual attributes and their dependencies
- added virtual attribute processor to act on membership status changes and group attribute changes
- added virtual user attribute "sponsoredMembershipInOrganizations"

### CODE REVIEW

**Review by zlamalp — CHANGES_REQUESTED** [review]
Please rebase this pull-request, I'll review it afterwards. Thank you.

**Comment by mvocu** [comment]
Rebased.

File: perun-ldapc/src/main/resources/perun-ldapc.xml [hunk 1]
```diff
@@ -676,6 +676,105 @@ http://www.springframework.org/schema/aop http://www.springframework.org/schema/
...
+					<property name="handlerMethodName" value="processGroupMemberChange"/>
+				</bean>
+				<bean class="cz.metacentrum.perun.ldapc.processor.impl.RegexpDispatchEventCondition">
+					<property name="beansCondition">
+						<list>
+							<value>cz.metacentrum.perun.core.api.Group</value>
+							<value>cz.metacentrum.perun.core.api.Member</value>
+						</list>
+					</property>
+					<property name="pattern" value="expired.$|disabled.$|invalidated.$|suspended #"/>
```
Comment (zlamalp) [inline perun-ldapc/src/main/resources/perun-ldapc.xml:725]: We recently changed member suspended message, so it should be ``suspended.$`` just like the other member states.

**Review by zlamalp — CHANGES_REQUESTED** [review]

File: perun-ldapc/src/main/resources/perun-ldapc.xml [hunk 2]
```diff
@@ -676,6 +676,105 @@ http://www.springframework.org/schema/aop http://www.springframework.org/schema/
...
+					<property name="pattern" value=" validated.$"/>
+					<property name="handlerMethodName" value="processMemberChange"/>
+				</bean>
+				<bean class="cz.metacentrum.perun.ldapc.processor.impl.RegexpDispatchEventCondition">
+					<property name="beansCondition">
+						<list>
+							<value>cz.metacentrum.perun.core.api.Member</value>
+						</list>
+					</property>
+					<property name="pattern" value="expired.$|disabled.$|invalidated.$|suspended #"/>
```
Comment (zlamalp) [inline perun-ldapc/src/main/resources/perun-ldapc.xml:743]: We recently changed member suspended message, so it should be ``suspended.$`` just like the other member states.

File: perun-ldapc/src/main/resources/perun-ldapc.xml (hunk 1 above)
Comment (mvocu) [reply perun-ldapc/src/main/resources/perun-ldapc.xml]: Fixed.

File: perun-ldapc/src/main/resources/perun-ldapc.xml (hunk 2 above)
Comment (mvocu) [reply perun-ldapc/src/main/resources/perun-ldapc.xml]: Fixed.

**Review by zlamalp — APPROVED** [review]

**Pull request merged**